src/core/CubicBezierLine.tsx

```diff
@@ -0,0 +1,30 @@
+import * as React from 'react'
+import { CubicBezierCurve3, Vector3 } from 'three'
+import { Line2 } from 'three/examples/jsm/lines/Line2'
+import { Line, LineProps } from './Line'
+
+type Props = Omit<LineProps, 'points'> & {
+  start: Vector3 | [number, number, number]
+  end: Vector3 | [number, number, number]
+  midA: Vector3 | [number, number, number]
+  midB: Vector3 | [number, number, number]
+  segments?: number
+}
+
+export const CubicBezierLine = React.forwardRef<Line2, Props>(function CubicBezierLine(
```

Comment:
Should Line2 not be Line? then you don't have to cast any

Reply:
Actually that's not the problem. The ref type should be the underlying Line2 from the Line component, not the Line component itself.

If I remove the `as any` cast I get this error:
```
Type 'Ref<ReactNode> | ((instance: Line2 | null) => void) | MutableRefObject<Line2 | null>' is not assignable to type 'Ref<Line2> | undefined'.
  Type 'RefObject<ReactNode>' is not assignable to type 'Ref<Line2> | undefined'.
    Type 'RefObject<ReactNode>' is not assignable to type 'RefObject<Line2>'.
      Type 'ReactNode' is not assignable to type 'Line2'.
        Type 'undefined' is not assignable to type 'Line2'.
```